### PR TITLE
[Merged by Bors] - feat: introducing the input data of the snake lemma

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+/-- `L₀.X₃` is the kernel of `v₁₂.τ₃ : L₁.X₃ ⟶ L₂.X₃`. -/
+noncomputable def h₀τ₃ : IsLimit (KernelFork.ofι S.v₀₁.τ₃ S.w₀₂_τ₃) :=
+  isLimitForkMapOfIsLimit' π₃ S.w₀₂ S.h₀
+
+instance mono_v₀₁_τ₁ : Mono S.v₀₁.τ₁ := mono_of_isLimit_fork S.h₀τ₁
+instance mono_v₀₁_τ₂ : Mono S.v₀₁.τ₂ := mono_of_isLimit_fork S.h₀τ₂
+instance mono_v₀₁_τ₃ : Mono S.v₀₁.τ₃ := mono_of_isLimit_fork S.h₀τ₃
+
+/-! The upper part of the first column of the snake diagram is exact. -/
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean:142]:
```suggestion
/-- The upper part of the first column of the snake diagram is exact. -/
```
I don't know if `/-!` does something, but I think this is a typo. Similar elsewhere.

File: Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+/-!
+# The snake lemma
+
+The snake lemma is a standard tool in homological algebra. The basic situation
+is when we have a diagram as follows in an abelian category `C`, with exact rows:
+
+    L₁.X₁ ⟶ L₁.X₂ ⟶ L₁.X₃ ⟶ 0
+      |       |       |
+      v       v       v
+0 ⟶ L₂.X₁ ⟶ L₂.X₂ ⟶ L₂.X₃
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean:18]:
```suggestion
    L₁.X₁ ⟶ L₁.X₂ ⟶ L₁.X₃ ⟶ 0
      |       |       |
      v.τ₁    v.τ₂   v.τ₃
0 ⟶ L₂.X₁ ⟶ L₂.X₂ ⟶ L₂.X₃
```
It helps to recall how this are named.

**Review by riccardobrasca — COMMENTED** [review]
LGTM, thanks!

bors d+

File: Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean]: Thanks!

File: Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean (hunk 2 above)
Comment (joelriou) [reply Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean]: Thanks!

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**